src/test/java/org/prebid/server/bidder/metax/MetaxBidderTest.java

```diff
@@ -0,0 +1,371 @@
+
+        //when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        //then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(HttpRequest::getImpIds)
+                .containsExactlyInAnyOrder(Collections.singleton("givenImp1"), Collections.singleton("givenImp2"));
+    }
+
+    @Test
+    public void shouldMakeOneRequestWhenOneImpIsValidAndAnotherIsNot() {
+        // given
+        final Imp givenInvalidImp = givenImp(imp -> imp
+                .id("impIdCorrupted")
+                .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()))));
+        final Imp givenValidImp = givenImp(identity());
+
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(List.of(givenInvalidImp, givenValidImp))
+                .build();
+
+        //when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        //then
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(HttpRequest::getPayload)
+                .flatExtracting(BidRequest::getImp)
+                .extracting(Imp::getId)
+                .containsExactly("123");
+    }
+
+    @Test
+    public void makeHttpRequestsShouldMakeOneRequestPerImp() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(givenImp(identity()), givenImp(identity())))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(HttpRequest::getPayload)
+                .extracting(BidRequest::getImp)
+                .extracting(List::size)
+                .containsOnly(1);
+    }
+
+    @Test
+    public void makeHttpRequestsShouldNotChangeBannerWidthAndHeightIfPresent() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(impCustomizer -> impCustomizer
+                .banner(Banner.builder()
+                        .w(12)
+                        .h(34)
+                        .format(singletonList(Format.builder().w(56).h(78).build()))
+                        .build())
+        );
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getPayload)
+                .flatExtracting(BidRequest::getImp)
+                .extracting(Imp::getBanner)
+                .extracting(Banner::getW, Banner::getH)
+                .containsExactly(Tuple.tuple(12, 34));
```

Comment:
I suggest asserting banner as a whole

Reply:
I like to check only specific fields, because when something changes on higher level I need to update not related tests

Reply:
imho at least it will be clear that something was changed and some tests are required